File: frontend/src/app/shared/model/dto/patent/patent.dto.ts
import { InstitutionDto } from '../institution.dto';
import { PatentAuthorDto } from './patent-author.dto';

export interface PatentDto {
  id: string;
  title: string;
  type: string;
  date: Date;
  authors: PatentAuthorDto[];
  institutes: InstitutionDto[];
}

File: backend_spring/tass-spring/src/main/java/pl/tass/tassspring/model/dto/patent/PatentDTO.java
package pl.tass.tassspring.model.dto.patent;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@AllArgsConstructor
@Data
@NoArgsConstructor
@Builder
public class PatentDTO {
    private String id;
    private String title;
    private String type;
    private LocalDate date;
    private List<PatentAuthorDTO> authors;
    private List<InstitutionDto> institutes;
}
